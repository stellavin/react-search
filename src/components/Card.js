import React from "react";

function Card({ movie }) {
  return (
    <div class="bg-white shadow-lg border-gray-100 max-h-44 border sm:rounded-3xl p-8 flex space-x-8 m-5 mb-20">
      <div class="h-48 overflow-visible w-3/2">
        <img
          class="rounded-3xl shadow-lg h-48"
          src="https://www.themoviedb.org/t/p/w600_and_h900_bestv2/1LRLLWGvs5sZdTzuMqLEahb88Pc.jpg"
          alt=""
        />
      </div>
      <div class="flex flex-col w-1/2">
        <div class="flex justify-between items-start">
          <h2 class="text-xl font-bold">Sweet Tooth: El ni√±o ciervo</h2>
        </div>
        <div>
          <div class="text-sm text-gray-400">Year</div>
          <div class="text-lg text-gray-800">2019</div>
        </div>
      </div>
    </div>
  );
}
export default Card;
