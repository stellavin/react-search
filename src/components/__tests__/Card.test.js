import { screen, cleanup, render } from "@testing-library/react";
import Card from "../Card";
import renderer from "react-test-renderer";

test("Should render Card Component with data", () => {
  const movie = {
    id: "a3614a1f-de11-4e15-8a41-50f8b01ba13c",
    name: "Spirited Away",
    description:
      "During her family's move to the suburbs, a sullen 10-year-old girl wanders into a world ruled by gods, witches, and spirits, and where humans are changed into beasts.",
    thumbnail:
      "https://m.media-amazon.com/images/M/MV5BNmU5OTQ0OWQtOTY0OS00Yjg4LWE1NDYtNDRhYWMxYWY4OTMwXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_UY268_CR3,0,182,268_AL_.jpg",
    rating: 8.6,
    duration: 7500,
    genres: ["animation", "adventure", "family"],
    releasedAt: "2003-03-27T22:00:00.000Z",
  };
  render(<Card movie={movie} />);
  const cardElement = screen.getByTestId("genre-1");
  expect(cardElement).toBeInTheDocument();
  expect(cardElement).toHaveTextContent("animation");
});

test("Card Component matches sanpshot", () => {
  const movie = {
    id: "a3614a1f-de11-4e15-8a41-50f8b01ba13c",
    name: "Spirited Away",
    description:
      "During her family's move to the suburbs, a sullen 10-year-old girl wanders into a world ruled by gods, witches, and spirits, and where humans are changed into beasts.",
    thumbnail:
      "https://m.media-amazon.com/images/M/MV5BNmU5OTQ0OWQtOTY0OS00Yjg4LWE1NDYtNDRhYWMxYWY4OTMwXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_UY268_CR3,0,182,268_AL_.jpg",
    rating: 8.6,
    duration: 7500,
    genres: ["animation", "adventure", "family"],
    releasedAt: "2003-03-27T22:00:00.000Z",
  };
  const tree = renderer.create(<Card movie={movie} />).toJSON;
  expect(tree).toMatchSnapshot();
});
