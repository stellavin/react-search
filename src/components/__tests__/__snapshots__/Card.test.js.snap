// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches sanpshot 1`] = `[Function]`;
